mlflow/pipelines/steps/train.py

```diff
@@ -114,6 +119,8 @@ def _run(self, output_directory):
         tags = {
             MLFLOW_SOURCE_TYPE: SourceType.to_string(SourceType.PIPELINE),
             MLFLOW_PIPELINE_TEMPLATE_NAME: self.step_config["template_name"],
+            MLFLOW_PIPELINE_PROFILE_NAME: self.step_config["initial_config_profile"],
+            MLFLOW_PIPELINE_STEP_NAME: self.step_config["initial_config_step"]
```

Comment:
Here you should be able to get the step name via `self.step_config["run_args"]["step"]`